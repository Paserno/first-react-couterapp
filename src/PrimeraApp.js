import React from 'react';
import PropTypes from 'prop-types';

// Funcional Coponent React

const PrimeraApp = ( {saludo, subtitulo} ) => {

    

    return (
        <>
            <h1>{ saludo }!</h1>
            {/* <pre>{JSON.stringify(saludo, null, 3)}</pre> */}
            <p>{subtitulo}</p>
        </>
    );
}
//! isRequired usar cuando es obligatorio el elemento
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}
PrimeraApp.defaultProps = {
    subtitulo:'Soy un subtitulo'
}

export default PrimeraApp;