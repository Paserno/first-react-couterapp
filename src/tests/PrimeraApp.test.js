import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import PrimeraApp from "../PrimeraApp";


describe('Prueba en <PrimeraApp />', () => {
    
    // test('debe de mostrar el mensaje "Hola, soy Paserno" ', () => {
        
    //     const saludo = 'Hola, soy Paserno';

    //     const {getByText} = render(<PrimeraApp saludo={saludo}/>);

    //     expect(getByText(saludo)).toBeInTheDocument();
    // });
    
    test('Debe de mostrar <PrimeraApp /> correctamente', () => {

        const saludo = 'Hola, soy Paserno';
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>);

        expect(wrapper).toMatchSnapshot();
    });
    
    
});
