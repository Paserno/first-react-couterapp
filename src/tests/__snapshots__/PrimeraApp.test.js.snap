// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <PrimeraApp /> Debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Paserno
    !
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
